### PR TITLE
First spring hw

### PR DESCRIPTION
**Pull request by AdamMudrak**

### CODE REVIEW

File: checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+        <module name="EmptyCatchBlock">
+            <property name="exceptionVariableName" value="expected"/>
+        </module>
+        <module name="CommentsIndentation"/>
+        <module name="DeclarationOrder"/>
+        <module name="SingleSpaceSeparator">
+            <property name="validateComments" value="true"/>
+        </module>
+    </module>
+</module>
```
Comment (okuzan) [inline checkstyle.xml]: Always look thru your PR, and add an empty line where you see this red sign

File: src/main/java/com/example/onlinebookshop/entities/Book.java [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @NotNull(message = "Title may not be null")
+    private String title;
+    @NotNull(message = "Author may not be null")
+    private String author;
+    @NotNull(message = "isbn may not be null")
+    @Column(unique = true)
+    private String isbn;
+    @NotNull(message = "Price may not be null")
```
Comment (okuzan) [inline src/main/java/com/example/onlinebookshop/entities/Book.java]: If you want to enforce the non-null constraint at the database level, it is recommended to use the @Column(nullable = false) annotation. This annotation instructs JPA to create the corresponding column in the database table with a NOT NULL constraint.

**Review by okuzan — CHANGES_REQUESTED** [review]

File: checkstyle.xml (hunk 1 above)
Comment (AdamMudrak) [reply checkstyle.xml]: I added an empty line, but I don't get what red sign you're talking about. Could you give me a screenshot?

Comment (okuzan) [reply checkstyle.xml]: If there's no empty line, GitHub shows this:
image
if there is one, it doesn't show it, if two - shows one and so on

Comment (AdamMudrak) [reply checkstyle.xml]: Right, thanks a lot! I get it now.

**Comment by AdamMudrak** [comment]
I would like to ask you to pay specific attention to my pom.xml. Is it well organised? Is there anything that could\should be improved?

File: pom.xml [hunk 3]
```diff
@@ -35,6 +80,53 @@
 				<groupId>org.springframework.boot</groupId>
 				<artifactId>spring-boot-maven-plugin</artifactId>
 			</plugin>
+
```
Comment (MAMentorFX) [inline pom.xml]:
```suggestion
```

File: pom.xml [hunk 4]
```diff
@@ -35,6 +80,53 @@
...
+						<path>
+							<groupId>org.projectlombok</groupId>
+							<artifactId>lombok-mapstruct-binding</artifactId>
+							<version>${lombok-mapstruct-binding.version}</version>
+						</path>
+					</annotationProcessorPaths>
+				</configuration>
+			</plugin>
 		</plugins>
 	</build>
```
Comment (MAMentorFX) [inline pom.xml]:
```suggestion
```

File: src/main/java/com/example/onlinebookshop/services/impl/BookServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
+    }
+
+    @Override
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
```
Comment (MAMentorFX) [inline src/main/java/com/example/onlinebookshop/services/impl/BookServiceImpl.java]: Mapstruct supports mapping collections from the box

**Review by MAMentorFX — CHANGES_REQUESTED** [review]
pom is good.

**Comment by AdamMudrak** [comment]
I found this example of mapping collections, hope it will suffice

**Review by Vasul-Pankiv — APPROVED** [review]

**Pull request merged**